main.nf

```diff
@@ -902,11 +903,14 @@ process featureCounts {
     } else if (reverse_stranded && !unstranded){
         featureCounts_direction = 2
     }
+    // Try to get real sample name
+    sample_name = bam_featurecounts.baseName - ~/Aligned.sortedByCoord.out/
```

Comment:
Yes! I found something! I don't think you need to use a regex here, just subtracting the string along _should_ work I think...

```groovy
sample_name = bam_featurecounts.baseName - 'Aligned.sortedByCoord.out'
```

(side note - this is a regex, so `.` will be matching any character. Including a literal `.`)